src/img/lib/sources/docker.js

```diff
@@ -162,6 +161,126 @@ function _importInfoFromV21Manifest(opts, cb) {
     cb(null, importInfo);
 }
 
+/*
+ * Handle manifest lists - select the appropriate manifest from the
+ * list of possible platforms, currently only fetches linux-amd64 or
+ * linux-386, that depends which arch is first on the manifest.
+ */
+function _getPlatformManifestfromList(opts, cb) {
+  assert.object(opts, 'opts');
+  assert.object(opts.dockerManifest, 'opts.dockerManifest');
+  assert.object(opts.rat, 'opts.rat');
+  assert.object(opts.client, 'opts.client');
+  assert.func(cb, 'cb');
+  assert.arrayOfObject(opts.dockerManifest.manifests);
+  if (!opts.dockerManifest.hasOwnProperty('mediaType')) {
+    cb(
+      new errors.InvalidDockerInfoError(
+        "No docker manifest mediaType defined"
+      )
+    );
+    return;
+  }
+
+  var supportedPlatforms = ['amd64', '386'];
+  var selectedDigest = null;
+  var rat = opts.rat;
+  var ManifestList = opts.dockerManifest.manifests;
+
+  for (var i = 0; i < ManifestList.length; i++) {
+      var p = ManifestList[i].platform;
+      if (p && p.os === "linux" &&
+          supportedPlatforms.indexOf(p.architecture) !== -1) {
+          selectedDigest = ManifestList[i].digest;
+          break;
+      }
+  }
+  if (!selectedDigest) {
+    var plats = [];
+    ManifestList.forEach(function _maniMap(mani) {
+      plats.push(
+        (mani.platform &&
+          format("%s-%s", mani.platform.os, mani.platform.architecture)) ||
+          "<unknown>"
+      );
+    });
+    cb(
+      new errors.InvalidDockerInfoError(
+        format("no linux-amd64 platform found in manifest list: %s", plats)
+      )
+    );
+    return;
+  }
+  var client = opts.client;
+  client.getManifest({ ref: selectedDigest },
+                      function (err, dockerManifest, res) {
+      if (err) {
+        if (err.statusCode === 404) {
+          cb();
+        } else {
+          cb(err);
+        }
+        return;
+      }
+      assert.arrayOfObject(dockerManifest.layers, 'dockerManifest.layers');
+      var layerDigests = dockerManifest.layers.map(function (l) {
+        return l.digest;
+      });
+
+      if (!dockerManifest.config.digest) {
+        cb(
+          new errors.InvalidDockerInfoError(
+            "No docker manifest config defined"
+          )
+        );
+        return;
+      }
+      var configDigest = dockerManifest.config.digest;
```

Comment:
Is this setup and call of createBlobreadStream() the same as the one below?  IF SO can it be refactored into a single function both here and the original one below can call?  I may be missing subtleties, however.